charmcraft/providers.py

```diff
@@ -171,26 +173,37 @@ def get_instance_name(
 
 def get_base_configuration(
     *,
-    alias: bases.BuilddBaseAlias,
+    alias: enum.Enum,
     instance_name: str,
-) -> bases.BuilddBase:
-    """Create a BuilddBase configuration."""
-    environment = get_command_environment()
-
+) -> Union[bases.ubuntu.BuilddBase, bases.centos.CentOSBase]:
+    """Create a Base configuration."""
     # injecting a snap on a non-linux system is not supported, so default to
     # install charmcraft from the store's stable channel
     snap_channel = get_managed_environment_snap_channel()
     if snap_channel is None and sys.platform != "linux":
         snap_channel = "stable"
 
-    charmcraft_snap = bases.buildd.Snap(name="charmcraft", channel=snap_channel, classic=True)
-    return bases.BuilddBase(
-        alias=alias,
-        environment=environment,
-        hostname=instance_name,
-        snaps=[charmcraft_snap],
-        compatibility_tag=f"charmcraft-{bases.BuilddBase.compatibility_tag}.0",
-    )
+    if isinstance(alias, bases.ubuntu.BuilddBaseAlias):
+        environment = get_command_environment(bases.ubuntu.BuilddBase)
+        charmcraft_snap = bases.ubuntu.Snap(name="charmcraft", channel=snap_channel, classic=True)
+        return bases.ubuntu.BuilddBase(
+            alias=alias,
+            environment=environment,
+            hostname=instance_name,
+            snaps=[charmcraft_snap],
+            compatibility_tag=f"charmcraft-{bases.ubuntu.BuilddBase.compatibility_tag}.0",
+        )
+
+    if isinstance(alias, bases.centos.CentOSBaseAlias):
```

Comment:
Can you add something to this function to handle when neither of the if statements are true?

Since it only returns inside `if` statements, perhaps a `raise Exception("base is not supported")` at the end of the function will do the trick.

In theory, this error case shouldn't occur with how this function is called, but it's still a good design practice.